app/scripts/metamask-controller.js

```diff
@@ -1061,6 +1065,11 @@ export default class MetamaskController extends EventEmitter {
 
     // TODO:LegacyProvider: Delete
     this.publicConfigStore = this.createPublicConfigStore();
+
+    // Multiple MetaMask instances launched warning
+    this.extension.runtime.onMessageExternal.addListener(onMessageReceived);
```

Comment:
This is probably challenging to test, but it would be great to confirm this is wired up correctly in the `metamask-controller` tests.

Reply:
Hmm not sure how valuable such test would be, but could you suggest the best way of looking into listeners to verify?

Reply:
I didn't get time to look into this, but I agree that maybe the benefit is minimal given how things are currently setup. We can tackle this later.